domain/service/employees.go

```diff
@@ -0,0 +1,187 @@
+package service
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"io"
+	"time"
+
+	"github.com/Shelex/grpc-go-demo/domain/entities"
+	"github.com/Shelex/grpc-go-demo/proto"
+	"github.com/Shelex/grpc-go-demo/storage"
+	"github.com/Shelex/grpc-go-demo/storage/documents"
+	"github.com/google/uuid"
+	log "github.com/sirupsen/logrus"
+	"google.golang.org/grpc"
+	"google.golang.org/grpc/metadata"
+)
+
+type employeeService struct {
+	repository storage.Storage
+	documents  documents.FileStorage
+	grpc       *grpc.Server
+}
+
+func (e *employeeService) Employees(req *proto.GetAllRequest, stream proto.EmployeeService_EmployeesServer) error {
+	employees, err := e.repository.GetAll()
+	if err != nil {
+		return fmt.Errorf("failed to get employees: %w", err)
+	}
+	for _, emp := range employees {
+		if err := stream.Send(&proto.EmployeeResponse{Employee: entities.EmployeeFromStorageToProto(emp)}); err != nil {
+			return err
+		}
+	}
+	return nil
+}
+
+func (e *employeeService) EmployeeByID(ctx context.Context, req *proto.ByIDRequest) (*proto.EmployeeResponse, error) {
+	if md, ok := metadata.FromIncomingContext(ctx); ok {
+		log.Printf("metadata received: %v\n", md)
+	}
+	employee, err := e.repository.GetEmployee(req.ID)
+	if err != nil {
+		return nil, err
+	}
+	return &proto.EmployeeResponse{
+		Employee: entities.EmployeeFromStorageToProto(employee),
+	}, nil
+}
+
+func (e *employeeService) AddEmployee(ctx context.Context, req *proto.EmployeeRequest) (*proto.EmployeeResponse, error) {
+	req.Employee.ID = uuid.New().String()
+	employee, err := e.repository.AddEmployee(entities.EmployeeFromProtoToStorage(req.Employee))
```

Comment:
golang style typically uses short abbreviated names, where it makes sense e.g. `emp`... I know though this doesnt always make sense